notebooks/noj_book/echarts.clj

```diff
@@ -189,23 +227,164 @@ data-for-waterfall
            :right "4%"
            :bottom "3%"
            :containLabel true}
-    :xAxis {:type "category"
+    :xAxis {:type :category
             :splitLine {:show false}
             :data (map #(str "Oct/" %) (range 1 12))}
-    :yAxis {:type "value"}
-    :series [{:type "bar"
+    :yAxis {:type :value}
+    :series [{:type :bar
               :stack "all"
               :itemStyle {:normal {:barBorderColor "rgba(0,0,0,0)"
                                    :color "rgba(0,0,0,0)"}
                           :emphasis {:barBorderColor "rgba(0,0,0,0)"
                                      :color "rgba(0,0,0,0)"}}
               :data suspended-cumulative}
              {:name "positive"
-              :type "bar"
+              :type :bar
               :stack "all"
               :data positive}
              {:name "negative"
-              :type "bar"
+              :type :bar
               :stack "all"
               :data negative
               :itemStyle {:color "#f33"}}]}))
+
+;; ### Line
+;; #### Dataset Preparation
+;; Before we plot any line charts, it's helpful to prepare the dataset first.
+
+(def stocks
```

Comment:
It would be great to add this definition to the `noj-book.datasets` namespace, so that other tutorials will be able to use it.

Reply:
I didn't know we could do it that way, so I am moving it there.